boa/src/syntax/ast/op.rs

```diff
@@ -51,32 +116,150 @@ impl Display for NumOp {
+    ///
+    /// Syntax: `-x`
+    ///
+    /// Converts non-numbers data types to numbers like unary plus,
+    /// however, it performs an additional operation, negation.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference](https://tc39.es/ecma262/#sec-unary-minus-operator)
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Unary_negation
     Minus,
-    /// `+a` - convert to a number
+
+    /// The unary plus operator attempts to convert the operand into a number, if it isn't already.
+    ///
+    /// Syntax: `+x`
+    ///
+    /// Although unary negation (`-`) also can convert non-numbers, unary plus is the fastest and preferred
+    /// way of converting something into a number, because it does not perform any other operations on the number.
+    /// It can convert `string` representations of integers and floats, as well as the non-string values `true`, `false`, and `null`.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference](https://tc39.es/ecma262/#sec-unary-plus-operator)
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Unary_plus
     Plus,
-    /// `!a` - get the opposite of the boolean value
+
+    /// Returns `false` if its single operand can be converted to `true`; otherwise, returns `true`.
+    ///
+    /// Syntax: `!x`
+    ///
+    /// Boolean values simply get inverted: `!true === false` and `!false === true`.
+    /// Non-boolean values get converted to boolean values first, then are negated.
+    /// This means that it is possible to use a couple of NOT operators in series to explicitly
+    /// force the conversion of any value to the corresponding boolean primitive.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference](https://tc39.es/ecma262/#sec-logical-not-operator)
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Logical_NOT
     Not,
-    /// `~a` - bitwise-not of the value
+
+    /// Performs the NOT operator on each bit.
+    ///
+    /// Syntax: `~x`
+    ///
+    /// NOT `a` yields the inverted value (or one's complement) of `a`.
+    /// Bitwise NOTing any number x yields -(x + 1). For example, ~-5 yields 4.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference](https://tc39.es/ecma262/#sec-bitwise-not-operator)
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators#Bitwise_NOT
     Tilde,
-    /// `typeof` - Get the type of object
+
+    /// The `typeof` operator returns a string indicating the type of the unevaluated operand.
+    ///
+    /// Syntax: `typeof expression` or `typeof(expression)`
```

Comment:
Since we are using this syntax above...
```suggestion
    /// Syntax: `typeof x` or `typeof(x)`
```